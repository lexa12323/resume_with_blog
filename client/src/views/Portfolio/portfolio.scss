.portfolio{
    display: flex;
    flex-wrap: wrap;
    gap: 2rem;
    margin: 30px 0;
    &__item{
        width: calc((100% / 3) - (4rem / 3));
    }
}