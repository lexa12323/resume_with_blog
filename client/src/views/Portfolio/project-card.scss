.project-card{
    &__image{
        width: 100%;
        height: 250px;
        opacity: 0.7;
        transition: opacity 0.3s ease-in-out;
    }
    &__name{
        font-weight: 700;
        font-size:18px;
        margin: 10px 0;
    }

    &__description {
        font-size:16px;
        margin: 10px 0;
    }

    &__link{
    }

    &:hover{
        .project-card__image{
            opacity: 1;
        }
    }
}