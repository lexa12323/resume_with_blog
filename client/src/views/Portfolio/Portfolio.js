import React from 'react'
import PropTypes from 'prop-types';
import './portfolio.scss'
import './project-card.scss'

export const PortfolioList = ({list}) => {
    return (
        <div className="portfolio">
            { list.map((item, index) => <PortfolioItem key={index} {...item}/>) }
        </div>
    )
}

export const PortfolioItem = ({image, name, link, description}) => {

    return (
        <div className="project-card portfolio__item">
            <img className="project-card__image" src={`/images/portfolio/${image}`} alt='portfolio'/>
            <div className="project-card__name">{name}</div> 
            <div className="project-card__description">{description}</div> 
            {link && <div className="project-card__link">{link}</div> }    
        </div>
    )

}

PortfolioList.propTypes = {
    list: PropTypes.array,
};

PortfolioItem.propTypes = {
    image: PropTypes.string,
    name: PropTypes.string,
    link: PropTypes.string,
    description: PropTypes.string,
};