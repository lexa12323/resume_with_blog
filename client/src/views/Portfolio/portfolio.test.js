import React from 'react'
import { PortfolioItem, PortfolioList } from './Portfolio'
//import { shallow } from 'enzyme';

describe('should render PortfolioItem component' , () => {
    it('should contain .portfolio__item item' , () => {
        const component = shallow(<PortfolioItem />);
        const wrapper = component.find('.portfolio__item')
        expect(wrapper.length).toBe(1);
    });
    
    it('should contain .project-card__link item' , () => {
        const component = shallow(<PortfolioItem link="#"/>);
        const wrapper = component.find('.project-card__link')
        expect(wrapper.text()).toBe('#');
    });
})

const data3 = [
    {image: 'gorbenko_resume.png', name: 'Сайт, на котором вы находитесь', description: 'Стек технологий: MERN'},
    {image: 'gorbenko_resume.png', name: 'Сайт, на котором вы находитесь', description: 'Стек технологий: MERN'},
]

describe('PortfolioList component' , () => {
    it('should render PortfolioList component' , () => {
        const component = shallow(<PortfolioList list={data3}/>);
        expect(component).toMatchSnapshot();
    });
})


/*describe('Props' , () => {
    it('should render PortfolioList component' , () => {
        const component = shallow(<PortfolioList list={data3}/>);
        expect(component).toMatchSnapshot();
    });
})*/