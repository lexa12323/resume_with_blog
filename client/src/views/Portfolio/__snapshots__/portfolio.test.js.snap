// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioList component should render PortfolioList component 1`] = `
<div
  className="portfolio"
>
  <PortfolioItem
    description="Стек технологий: MERN"
    image="gorbenko_resume.png"
    key="0"
    name="Сайт, на котором вы находитесь"
  />
  <PortfolioItem
    description="Стек технологий: MERN"
    image="gorbenko_resume.png"
    key="1"
    name="Сайт, на котором вы находитесь"
  />
</div>
`;
